cli/Cargo.toml

```diff
@@ -99,6 +99,7 @@ shell-escape = "=0.1.5"
 tar.workspace = true
 text-size = "=1.1.0"
 text_lines = "=0.6.0"
+thiserror = "=1.0.38"
```

Comment:
Not a fan of introducing another error crate, but if you feel it's beneficial then okay. (Maybe we should start migrating away from `anyhow`)

Reply:
Apparently the developer of `anyhow` has some other errorlib (maybe precisely `thiserror`?) that they recommend nowadays. Can't remember what the reason was but this was mentioned in a Rust error handling talk in a meetup I attended this past November.

Reply:
Yeah, it's the compliment to anyhow. I made it use thiserror when moving to deno_graph and I didn't bother unrefactoring the code here because Luca wanted us to start using it in the non-CLI crates more, which I think makes sense. See #17318 for other discussion too.